Source/Core/Core/AchievementManager.h

```diff
@@ -80,6 +81,7 @@ class AchievementManager
   std::map<AchievementId, UnlockStatus> m_unlock_map;
 
   Common::WorkQueueThread<std::function<void()>> m_queue;
+  std::mutex m_lock;
```

Comment:
`std::recursive_mutex` is generally the better choice so that you can call methods that lock while you're already locked.